lib/pgpool_no_load_balance.rb

```diff
@@ -12,7 +12,7 @@ module PgpoolNoLoadBalance
   class PostgreSQLAdapterMissing < StandardError; end
 
   def self.setup!
-    unless ::ActiveRecord::Base.respond_to?(:postgresql_connection)
+    unless defined?(::ActiveRecord::ConnectionAdapters::PostgreSQLAdapter)
```

Comment:
This is what’s failing due to the Rails 7.2.x upgrade; the other updated files are from the Rails 7.1 branch.